### ISSUE TITLE
Modify widgets after creation?

### ISSUE THREAD

**Issue by kapily** [mljar/mercury#285]
Looking at the example of "dynamic widgets" here:
https://runmercury.com/examples/dynamic-widgets/

It seems possible to change a widget based on the output of a _previous_ widget. But what if you wanted to change the output of a widget based on the current value of the widget? 

This would be very useful in cases like validation and formatting. For example, if you have a "date" input, you might want to standardize the formatting of the date into the format `yyyy-mm-dd`, even if the user enters `yyyymmdd`

In order to do this, it would be good if you supported setting the value of a widget after instantiation (b/c when you read the `.text` of the field, you still receive the last field the user inputted)

**Comment by kapily** [comment]
It seems like the setter was added at one time, but is commented out:
https://github.com/mljar/mercury/blob/10420e73b6dfbb7ddaba0ea2e72604337ffe36e8/mercury/widgets/text.py#L33

But the way it's written, it seems like the setter would not work? I think the widget needs to be refreshed after setting the value.

**Comment by kapily** [comment]
I tried monkey patching `Text`, but this didn't work:

I was trying to monkey patch this:
https://github.com/mljar/mercury/blob/main/mercury/widgets/text.py

via:
```python
import mercury as mr

import ipywidgets
from IPython.display import display
from mercury import WidgetsManager

def text_init(self, value="", label="", rows=1, url_key="", disabled=False, hidden=False, should_display=True, update_value=False, value_fn=None):
    self.rows = rows
    self.code_uid = WidgetsManager.get_code_uid("Text")
    self.url_key = url_key
    self.hidden = hidden
    self._displayed = False
    if WidgetsManager.widget_exists(self.code_uid):
        self.text = WidgetsManager.get_widget(self.code_uid)
        self.text.description = label
        self.text.disabled = disabled
        if update_value and self.text.value != value:
            self.text.value = value
        if value_fn and self.text.value != value_fn(self.text.value):
            self.text.value = value_fn(self.text.value)
        # What I noticed is that the print statement below printed out the correct value that
        # expected. However, the text field in the actual UI did not update
        print('text value should now be: ', self.text.value)
    else:
        if value_fn:
            value = value_fn(value)
        self.text = ipywidgets.Textarea(
            value=value, description=label, disabled=disabled
        )
        WidgetsManager.add_widget(self.text.model_id, self.code_uid, self.text)
    if should_display:
        self.manual_display()

def manual_display(self):
    assert not self._displayed
    self._displayed = True
    display(self)

mr.Text.__init__ = text_init
mr.Text.manual_display = manual_display
```

but it didn't correctly update the text in the UI. 
I added the following fields:
`should_display=True, update_value=False, value_fn=None`
as various means to update the widget after creation. But again, as mentioned, it did not update the text in the Text widget.

**Comment by pplonski** [comment]
Widget value can be only changed in UI - it is by design. 

I need some time to think about option to allow user set the initial value of the widget. If your widget wasn't updated in the UI then we can allow to change the `value` in the code. So you will be able to dynamically set widget value, and the below code will work (currently it will not change the `text.value`).
![image](https://github.com/mljar/mercury/assets/6959032/12fc847f-20c9-4b06-bfd9-a92889e31085)

What do you think?

**Comment by kapily** [comment]
Basically I was hoping to be able to change the widget value even after it was edited. This is useful for data validation, etc. What about another parameter to basically overwrite the existing value `overwrite_value=False` that's by default set to false (current behavior)

**Comment by kapily** [comment]
Yes - agree that by default, we do not want the UI to over-write it. But in cases like:
- data validation
- having a `select` change the value of a input text field

it would be good to be able to programmatically override it. I think the `overwrite_value` allows that ability without changing existing behavior. You could even create a baseclass that all widgets subclass off of and add it in there.
